NullPointer exception when parsing the baseline yaml

We noticed a NPE during baseline processing phase in case a declared module has no suppressions. We entered into the issue after manually adapting the baseline file during a refactoring and not via `projectGuardBaseline`

```
Caused by: java.lang.NullPointerException: Cannot invoke "java.lang.Iterable.iterator()" because "$this$forEach$iv" is null
	at com.rubensousa.projectguard.plugin.internal.SuppressionMap.set(SuppressionMap.kt:78)
	at com.rubensousa.projectguard.plugin.internal.task.CheckExecutor.execute-d1pmJ48(TaskCheck.kt:95)
	at com.rubensousa.projectguard.plugin.internal.task.TaskCheck.projectGuardCheck(TaskCheck.kt:73)
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,6 +1,10 @@
 
 # CHANGELOG
 
+## 1.0.0-rc01 (2026-03-14)
+
+- Fixed NPE caused by empty module suppressions when the baseline is manually edited 
+
 ## 1.0.0-beta04 (2026-03-06)
 
 - Added option to include `projectGuardCheck` as part of the build tasks with the following:

--- a/projectguard/src/main/kotlin/com/rubensousa/projectguard/plugin/internal/BaselineConfiguration.kt
+++ b/projectguard/src/main/kotlin/com/rubensousa/projectguard/plugin/internal/BaselineConfiguration.kt
@@ -17,5 +17,11 @@
 package com.rubensousa.projectguard.plugin.internal
 
 internal data class BaselineConfiguration(
-    val suppressions: Map<String, List<DependencySuppression>>
-)
+    val suppressions: Map<String, List<DependencySuppression>?>
+) {
+
+    fun getModuleSuppressions(moduleId: String): List<DependencySuppression> {
+        return suppressions[moduleId].orEmpty()
+    }
+
+}

--- a/projectguard/src/main/kotlin/com/rubensousa/projectguard/plugin/internal/BaselineProcessor.kt
+++ b/projectguard/src/main/kotlin/com/rubensousa/projectguard/plugin/internal/BaselineProcessor.kt
@@ -18,21 +18,26 @@ package com.rubensousa.projectguard.plugin.internal
 
 import com.fasterxml.jackson.databind.ObjectMapper
 import com.fasterxml.jackson.dataformat.yaml.YAMLFactory
+import com.fasterxml.jackson.module.kotlin.KotlinFeature
 import com.fasterxml.jackson.module.kotlin.KotlinModule
 import java.io.File
 
-internal class YamlProcessor {
+internal class BaselineProcessor {
 
     private val objectMapper = ObjectMapper(YAMLFactory())
         .apply {
-            registerModule(KotlinModule.Builder().build())
+            registerModule(
+                KotlinModule.Builder()
+                    .enable(KotlinFeature.NullToEmptyMap)
+                    .build()
+            )
         }
 
-    fun <T> parse(file: File, clazz: Class<T>): T {
-        return objectMapper.readValue(file, clazz)
+    fun parse(file: File): BaselineConfiguration {
+        return objectMapper.readValue(file, BaselineConfiguration::class.java)
     }
 
-    fun <T> write(file: File, content: T) {
+    fun write(file: File, content: BaselineConfiguration) {
         return objectMapper.writeValue(file, content)
     }
 

--- a/projectguard/src/main/kotlin/com/rubensousa/projectguard/plugin/internal/SuppressionMap.kt
+++ b/projectguard/src/main/kotlin/com/rubensousa/projectguard/plugin/internal/SuppressionMap.kt
@@ -24,8 +24,7 @@ internal class SuppressionMap {
         suppressions.clear()
         configuration.suppressions.forEach { entry ->
             val moduleId = entry.key
-            val suppressions = entry.value
-            suppressions.forEach { suppression ->
+            configuration.getModuleSuppressions(moduleId).forEach { suppression ->
                 add(moduleId, suppression)
             }
         }

--- a/projectguard/src/main/kotlin/com/rubensousa/projectguard/plugin/internal/task/TaskBaseline.kt
+++ b/projectguard/src/main/kotlin/com/rubensousa/projectguard/plugin/internal/task/TaskBaseline.kt
@@ -16,9 +16,8 @@
 
 package com.rubensousa.projectguard.plugin.internal.task
 
-import com.rubensousa.projectguard.plugin.internal.BaselineConfiguration
+import com.rubensousa.projectguard.plugin.internal.BaselineProcessor
 import com.rubensousa.projectguard.plugin.internal.SuppressionMap
-import com.rubensousa.projectguard.plugin.internal.YamlProcessor
 import com.rubensousa.projectguard.plugin.internal.report.RestrictionDump
 import kotlinx.serialization.json.Json
 import org.gradle.api.DefaultTask
@@ -58,14 +57,14 @@ internal class BaselineExecutor(
 
     fun execute() {
         val restrictionDump = Json.decodeFromString<RestrictionDump>(inputFile.readText())
-        val yamlProcessor = YamlProcessor()
+        val baselineProcessor = BaselineProcessor()
         val currentReasons = mutableMapOf<String, MutableMap<String, String>>()
         runCatching {
-            val currentConfiguration = yamlProcessor.parse(outputFile, BaselineConfiguration::class.java)
-            currentConfiguration.suppressions.forEach { (module, dependencies) ->
-                dependencies.forEach { (dependency, reason) ->
+            val currentConfiguration = baselineProcessor.parse(outputFile)
+            currentConfiguration.suppressions.keys.forEach { module ->
+                currentConfiguration.getModuleSuppressions(module).forEach { suppression ->
                     val dependencyReasons = currentReasons.getOrPut(module) { mutableMapOf() }
-                    dependencyReasons[dependency] = reason
+                    dependencyReasons[suppression.dependency] = suppression.reason
                 }
             }
         }
@@ -81,7 +80,7 @@ internal class BaselineExecutor(
                 )
             }
         }
-        yamlProcessor.write(outputFile, suppressionMap.getBaseline())
+        baselineProcessor.write(outputFile, suppressionMap.getBaseline())
     }
 
 }

--- a/projectguard/src/main/kotlin/com/rubensousa/projectguard/plugin/internal/task/TaskCheck.kt
+++ b/projectguard/src/main/kotlin/com/rubensousa/projectguard/plugin/internal/task/TaskCheck.kt
@@ -16,12 +16,11 @@
 
 package com.rubensousa.projectguard.plugin.internal.task
 
-import com.rubensousa.projectguard.plugin.internal.BaselineConfiguration
+import com.rubensousa.projectguard.plugin.internal.BaselineProcessor
 import com.rubensousa.projectguard.plugin.internal.DependencyGraphBuilder
 import com.rubensousa.projectguard.plugin.internal.ProjectGuardSpec
 import com.rubensousa.projectguard.plugin.internal.ReportSpec
 import com.rubensousa.projectguard.plugin.internal.SuppressionMap
-import com.rubensousa.projectguard.plugin.internal.YamlProcessor
 import com.rubensousa.projectguard.plugin.internal.report.DependencyGraphDump
 import com.rubensousa.projectguard.plugin.internal.report.HtmlReportGenerator
 import com.rubensousa.projectguard.plugin.internal.report.RestrictionDump
@@ -87,10 +86,10 @@ internal class CheckExecutor(
     fun execute(): Result<Unit> = runCatching {
         val dependencyGraphDump = Json.decodeFromString<DependencyGraphDump>(dependenciesFile.readText())
         val graph = DependencyGraphBuilder().buildFromDump(dependencyGraphDump)
-        val yamlProcessor = YamlProcessor()
+        val baselineProcessor = BaselineProcessor()
         val suppressionMap = SuppressionMap()
         runCatching {
-            yamlProcessor.parse(baselineFile, BaselineConfiguration::class.java)
+            baselineProcessor.parse(baselineFile)
         }.onSuccess { config ->
             suppressionMap.set(config)
         }.onFailure {

--- a/projectguard/src/test/kotlin/com/rubensousa/projectguard/plugin/TaskBaselineTest.kt
+++ b/projectguard/src/test/kotlin/com/rubensousa/projectguard/plugin/TaskBaselineTest.kt
@@ -18,8 +18,8 @@ package com.rubensousa.projectguard.plugin
 
 import com.google.common.truth.Truth.assertThat
 import com.rubensousa.projectguard.plugin.internal.BaselineConfiguration
+import com.rubensousa.projectguard.plugin.internal.BaselineProcessor
 import com.rubensousa.projectguard.plugin.internal.DependencySuppression
-import com.rubensousa.projectguard.plugin.internal.YamlProcessor
 import org.junit.Rule
 import org.junit.Test
 import org.junit.rules.TemporaryFolder
@@ -58,8 +58,8 @@ class TaskBaselineTest {
         val outputFile = plugin.generateBaseline()
 
         // then
-        val yamlProcessor = YamlProcessor()
-        val baseline = yamlProcessor.parse(outputFile, BaselineConfiguration::class.java)
+        val baselineProcessor = BaselineProcessor()
+        val baseline = baselineProcessor.parse(outputFile)
         assertThat(baseline.suppressions).isEqualTo(
             mapOf(
                 moduleId to listOf(
@@ -81,7 +81,7 @@ class TaskBaselineTest {
     @Test
     fun `baseline generation keeps the previous suppresion reasons`() {
         // given
-        val yamlProcessor = YamlProcessor()
+        val baselineProcessor = BaselineProcessor()
         val moduleId = ":domain"
         val secondModuleId = ":data"
         val fatalModuleId = ":legacy"
@@ -104,7 +104,7 @@ class TaskBaselineTest {
                 )
             )
         )
-        yamlProcessor.write(baselineFile, currentBaseline)
+        baselineProcessor.write(baselineFile, currentBaseline)
         plugin.dumpDependencies(moduleId) {
             addInternalDependency(moduleId, fatalModuleId)
         }
@@ -121,7 +121,7 @@ class TaskBaselineTest {
         val outputFile = plugin.generateBaseline()
 
         // then
-        val baseline = yamlProcessor.parse(outputFile, BaselineConfiguration::class.java)
+        val baseline = baselineProcessor.parse(outputFile)
         assertThat(baseline.suppressions).isEqualTo(
             mapOf(
                 moduleId to listOf(

--- a/projectguard/src/test/kotlin/com/rubensousa/projectguard/plugin/TaskCheckTest.kt
+++ b/projectguard/src/test/kotlin/com/rubensousa/projectguard/plugin/TaskCheckTest.kt
@@ -18,8 +18,8 @@ package com.rubensousa.projectguard.plugin
 
 import com.google.common.truth.Truth.assertThat
 import com.rubensousa.projectguard.plugin.internal.BaselineConfiguration
+import com.rubensousa.projectguard.plugin.internal.BaselineProcessor
 import com.rubensousa.projectguard.plugin.internal.DependencySuppression
-import com.rubensousa.projectguard.plugin.internal.YamlProcessor
 import org.junit.Rule
 import org.junit.Test
 import org.junit.rules.TemporaryFolder
@@ -155,7 +155,7 @@ class TaskCheckTest {
     @Test
     fun `task fails if baseline contains outdated references no longer part of the graph`() {
         // given
-        val yamlProcessor = YamlProcessor()
+        val baselineProcessor = BaselineProcessor()
         val moduleId = ":domain"
         val fatalModuleId = ":legacy"
         val baselineFile = plugin.getBaselineFile()
@@ -164,7 +164,7 @@ class TaskCheckTest {
                 moduleId to listOf(DependencySuppression(dependency = fatalModuleId)),
             )
         )
-        yamlProcessor.write(baselineFile, currentBaseline)
+        baselineProcessor.write(baselineFile, currentBaseline)
         plugin.dumpDependencies(moduleId) {
             addInternalDependency(moduleId, fatalModuleId)
         }

--- a/projectguard/src/test/kotlin/com/rubensousa/projectguard/plugin/internal/BaselineProcessorTest.kt
+++ b/projectguard/src/test/kotlin/com/rubensousa/projectguard/plugin/internal/BaselineProcessorTest.kt
@@ -21,12 +21,12 @@ import org.junit.Rule
 import org.junit.rules.TemporaryFolder
 import kotlin.test.Test
 
-class YamlProcessorTest {
+class BaselineProcessorTest {
 
     @get:Rule
     val temporaryFolder = TemporaryFolder()
 
-    private val yamlProcessor = YamlProcessor()
+    private val baselineProcessor = BaselineProcessor()
 
     @Test
     fun `suppression list is written and parsed back`() {
@@ -38,12 +38,42 @@ class YamlProcessorTest {
         val file = temporaryFolder.newFile("suppressions.yml")
 
         // when
-        yamlProcessor.write(file, configuration)
+        baselineProcessor.write(file, configuration)
 
         // then
-        val parsedConfiguration = yamlProcessor.parse(file, BaselineConfiguration::class.java)
+        val parsedConfiguration = baselineProcessor.parse(file)
         assertThat(parsedConfiguration).isEqualTo(configuration)
     }
 
+    @Test
+    fun `suppression without any module dependency is parsed correctly`() {
+        // given
+        val file = temporaryFolder.newFile("suppressions.yml")
+        file.writeText("""
+            suppressions:
+              module:
+        """.trimIndent())
+
+        // when
+        val parsedConfiguration = baselineProcessor.parse(file)
+
+        // then
+        assertThat(parsedConfiguration.getModuleSuppressions("module")).isEmpty()
+    }
+
+    @Test
+    fun `suppression without any module is parsed correctly`() {
+        // given
+        val file = temporaryFolder.newFile("suppressions.yml")
+        file.writeText("""
+            suppressions:
+        """.trimIndent())
+
+        // when
+        val parsedConfiguration = baselineProcessor.parse(file)
+
+        // then
+        assertThat(parsedConfiguration.suppressions).isEmpty()
+    }
 
 }

--- a/projectguard/src/test/kotlin/com/rubensousa/projectguard/plugin/internal/SuppressionMapTest.kt
+++ b/projectguard/src/test/kotlin/com/rubensousa/projectguard/plugin/internal/SuppressionMapTest.kt
@@ -89,11 +89,15 @@ class SuppressionMapTest {
 
         // then
         val entries = baseline.suppressions.entries.toList()
-        assertThat(entries[0].key).isEqualTo(":app")
-        assertThat(entries[0].value[0].dependency).isEqualTo(":a")
-        assertThat(entries[0].value[1].dependency).isEqualTo(":b")
-        assertThat(entries[1].key).isEqualTo(":lib")
-        assertThat(entries[1].value[0].dependency).isEqualTo(":c")
+        val firstEntry = entries[0]
+        val firstEntryDependencies = baseline.getModuleSuppressions(firstEntry.key)
+        val secondEntry = entries[1]
+        val secondEntryDependencies = baseline.getModuleSuppressions(secondEntry.key)
+        assertThat(firstEntry.key).isEqualTo(":app")
+        assertThat(firstEntryDependencies[0].dependency).isEqualTo(":a")
+        assertThat(firstEntryDependencies[1].dependency).isEqualTo(":b")
+        assertThat(secondEntry.key).isEqualTo(":lib")
+        assertThat(secondEntryDependencies[0].dependency).isEqualTo(":c")
     }
 
     @Test

